src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs

```diff
@@ -0,0 +1,1044 @@
+                            if (!thisPtr.AuthenticateContinue())
+                            {
+                                return;
+                            }
+                        }
+                        else if ((int)info.StatusCode == 235)
+                        {
+                            thisPtr._connection._authenticationModules[thisPtr._currentModule].CloseContext(thisPtr._connection);
+                            thisPtr._connection._isConnected = true;
+                            thisPtr.InvokeCallback();
+                            return;
+                        }
+
+                        thisPtr.Authenticate();
+                    }
+                    catch (Exception e)
+                    {
+                        thisPtr.InvokeCallback(e);
+                    }
+                }
+            }
+
+            private bool AuthenticateContinue()        //10
+            {
+                for (;;)
+                {
+                    // We don't need credential on the continued auth assuming they were captured on the first call.
+                    // That should always work, otherwise what if a new credential has been returned?
+                    Authorization auth = _connection._authenticationModules[_currentModule].Authenticate(_authResponse, null, _connection, _connection._client.TargetName, _connection._channelBindingToken);
+                    if (auth == null)
+                    {
+                        throw new SmtpException(SR.Format(SR.SmtpAuthenticationFailed));
+                    }
+
+                    IAsyncResult result = AuthCommand.BeginSend(_connection, auth.Message, s_authenticateContinueCallback, this);
+                    if (!result.CompletedSynchronously)
+                    {
+                        return false;
+                    }
+
+                    LineInfo info = AuthCommand.EndSend(result);
+                    if ((int)info.StatusCode == 235)
+                    {
+                        _connection._authenticationModules[_currentModule].CloseContext(_connection);
+                        _connection._isConnected = true;
+                        InvokeCallback();
+                        return false;
+                    }
+                    else if ((int)info.StatusCode != 334)
+                    {
+                        return true;
+                    }
+                    _authResponse = info.Line;
+                }
+            }
+
+            private static void AuthenticateContinueCallback(IAsyncResult result)     //11
+            {
+                if (!result.CompletedSynchronously)
+                {
+                    ConnectAndHandshakeAsyncResult thisPtr = (ConnectAndHandshakeAsyncResult)result.AsyncState;
+                    try
+                    {
+                        LineInfo info = AuthCommand.EndSend(result);
+                        if ((int)info.StatusCode == 235)
```

Comment:
SmtpStatusCode is a public enum, adding to it is a new API change. Out of scope of this PR and netstandard2.0